e2e/app/admin/config.go

```diff
@@ -1,10 +1,38 @@
 package admin
 
-type Config struct {
+import (
+	"context"
+)
+
+type BaseConfig struct {
+	// Broadcast  determines whether or not transactions.
+	Broadcast bool
+}
+
+type ChainConfig struct {
 	// Chain is the Name of chain to run on. Leave empty to run on all applicable chains.
 	Chain string
 }
 
-func DefaultConfig() Config {
-	return Config{Chain: ""}
+func DefaultChainConfig() ChainConfig {
+	return ChainConfig{Chain: ""}
+}
+
+func DefaultBaseConfig() BaseConfig {
+	return BaseConfig{Broadcast: false}
+}
+
+type configKey struct{}
+
+func CtxWithBaseConfig(ctx context.Context, cfg *BaseConfig) context.Context {
```

Comment:
putting critical info in the context is considered bad practice. Only stuff that you don't actually need for logic (nice-to-haves) should be included in ctx.

Reply:
yeah I was gonna ask you about this live.  it's nice not to have to wire it though all the function calls. and I thought requiring something be explicitly "positive" might make this okay? 

but yes, noted on base practices